src/hooks/useSignTransaction.ts

```diff
@@ -1,23 +1,43 @@
-import { useCallback } from 'react';
+import { useCallback, useEffect, useState } from "react";
 
-import { useCreateWalletClient } from '@/hooks/useCreateWalletClient';
+import { useCreateWalletClient } from "@/hooks/useCreateWalletClient";
+import { getTransactionCount } from "wagmi/actions";
+import { wagmiConfig } from "@/components/Web3ModalProvider";
 
 export const useSignTransaction = () => {
   const client = useCreateWalletClient();
+  const [transactionCount, setTransactionCount] = useState(0)
 
-  return useCallback(async (request: any) => {
-    // const request = await client.prepareTransactionRequest({
-    //   account: client.account,
-    //   to: '0x70997970c51812dc3a010c7d01b50e0d17dc79c8',
-    //   value: 1000000000000n
-    // })
+  useEffect(() => {
+    const fetchTransactionCount = async () => {
```

Comment:
Let's discuss potential avoid extra fetch nonce in Slack. Cause it could be got from prepareRequest hook